### PR TITLE
chore: refactor SQL operations unit tests with proptest

### PR DESCRIPTION
**Pull request by chloeminkyung**
Refactor existing unit tests with proptest
- [x] [number](https://github.com/getdozer/dozer/blob/main/dozer-sql/src/pipeline/expression/scalar/number.rs)
- [x] [mathematical](https://github.com/getdozer/dozer/blob/main/dozer-sql/src/pipeline/expression/scalar/tests/cast.rs#L646) // TODO: overflowing of Decimal for Multiplication & Division needs to be solved
- [x] [scalar](https://github.com/getdozer/dozer/blob/main/dozer-sql/src/pipeline/expression/scalar/number.rs)
- [x] [logical](https://github.com/getdozer/dozer/blob/main/dozer-sql/src/pipeline/expression/logical.rs#L71)
- [x] [datetime](https://github.com/getdozer/dozer/blob/main/dozer-sql/src/pipeline/expression/scalar/tests/datetime.rs)
- [x] [string](https://github.com/getdozer/dozer/blob/main/dozer-sql/src/pipeline/expression/scalar/string.rs)
- [x] [comparison](https://github.com/getdozer/dozer/blob/main/dozer-sql/src/pipeline/expression/comparison.rs#L373)
- [x] [geo](https://github.com/getdozer/dozer/tree/main/dozer-sql/src/pipeline/expression/geo/tests) // TODO: should be able to calculate `vincenty_distance` needs to be solved

Todo
- [ ] logical operator tests (starting from SQL, for example, using `run_scalar_fct()`) into dozer-tests
- [ ] [cast](https://github.com/getdozer/dozer/blob/main/dozer-sql/src/pipeline/expression/scalar/tests/cast.rs#L16) // TODO: only logical testing is possible

### CODE REVIEW

File: dozer-sql/Cargo.toml [hunk 1]
```diff
@@ -22,6 +22,7 @@ hashbrown = "0.13"
 ahash = "0.8.3"
 bloom = "0.3.2"
 enum_dispatch = "0.3.11"
+proptest = "1.1.0"
```
Comment (v3g42) [inline dozer-sql/Cargo.toml:25]: Can we move this to `dev-dependencies `?

Comment (chloeminkyung) [reply dozer-sql/Cargo.toml:25]: can

**Comment by v3g42** [comment]
looks good overall. Thanks @chloeminkyung 

I was wondering if there is a way to generate even the `FieldType` and `Field` and pump them through each function. And compare results using `sql_tests`. This could be a separate exercise.

**Review by v3g42 — APPROVED** [review]

File: dozer-sql/src/pipeline/expression/scalar/tests/mod.rs [hunk 2]
```diff
@@ -3,6 +3,8 @@ mod cast;
 #[cfg(test)]
 mod datetime;
 #[cfg(test)]
+mod mathematical;
```
[the remark this answers is not in the document]
Comment (chloeminkyung) [reply dozer-sql/src/pipeline/expression/scalar/tests/mod.rs:6]: it can't be it's a test not the module itself

**Pull request merged**